src/bin/vip-validate.js

```diff
@@ -0,0 +1,417 @@
+#!/usr/bin/env node
+// @flow
+
+/**
+ * External dependencies
+ */
+import {
+	Harmonia,
+	SiteConfig,
+	EnvironmentVariables,
+	TestSuite,
+	TestSuiteResult,
+	Test,
+	TestResult,
+	TestResultType,
+	IssueType,
+} from '@automattic/vip-go-preflight-checks';
+
+import path from 'path';
+import gql from 'graphql-tag';
+import { readFileSync } from 'fs';
+import dotenv from 'ini';
+import chalk from 'chalk';
+
+/**
+ * Internal dependencies
+ */
+import command from 'lib/cli/command';
+import * as exit from 'lib/cli/exit';
+import { default as API, enableGlobalGraphQLErrorHandling, disableGlobalGraphQLErrorHandling } from '../lib/api';
+
+export const appQuery = `
+	id
+	name
+	repo
+	environments {
+		id
+		appId
+		name
+		type
+		environmentVariables {
+			nodes {
+				name,
+				value
+			}
+		}
+	}
+	organization {
+		id
+		name
+	}
+`;
+
+let suppressOutput = false;
+let outputJson = false;
+
+function logToConsole( ...messages: string[] ) {
+	if ( suppressOutput ) {
+		return;
+	}
+
+	if ( messages.length === 0 ) {
+		messages = [ '' ];
+	}
+
+	messages.forEach( message => console.log( message ) );
+}
+
+async function getBuildConfiguration( application, environment ) {
+	const api = await API();
+
+	// Disable the global GraphQL error handling, so we can catch Unauthorized errors and recommend next steps.
+	disableGlobalGraphQLErrorHandling();
+
+	const buildConfigQuery = gql`
+	query BuildConfig( $appId: Int, $envId: Int ) {
+        app(id: $appId) {
+            environments(id: $envId) {
+                id,
+                buildConfiguration {
+					buildType
+					nodeBuildDockerEnv,
+					nodeJSVersion,
+					npmToken,
+                }
+            }
+        }
+	}`;
+
+	try {
+		const result = await api.query( {
+			query: buildConfigQuery,
+			fetchPolicy: 'network-only',
+			variables: {
+				appId: environment.appId,
+				envId: environment.id,
+			},
+		} );
+
+		// Reenable GraphQL error handling
+		enableGlobalGraphQLErrorHandling();
+
+		return result.data.app.environments[ 0 ].buildConfiguration;
+	} catch ( error ) {
+		if ( error.graphQLErrors && error.graphQLErrors.find( gqlError => gqlError.message === 'Unauthorized' ) ) {
+			console.log( `${ chalk.red( 'Error:' ) } You don't have the required permissions to run validations for this app, and to create the testing environment.\n` +
```

Comment:
Technically, they are targeting an "environment" not an "app." The use of "testing environment" is confusing and can be omitted.

```suggestion
			console.log( `${ chalk.red( 'Error:' ) } You don't have the required permissions to run validations for this environment.\n` +
```